tokio/tests/time_pause.rs

```diff
@@ -3,8 +3,29 @@
 
 use rand::SeedableRng;
 use rand::{rngs::StdRng, Rng};
-use tokio::time::{self, Duration, Instant};
-use tokio_test::assert_err;
+use tokio::time::{self, Duration, Instant, Sleep};
+use tokio_test::{assert_err, assert_pending, assert_ready_eq, task};
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+macro_rules! assert_elapsed {
+    ($now:expr, $ms:expr) => {{
+        let elapsed = $now.elapsed();
+        let lower = ms($ms);
+
+        // Handles ms rounding
+        assert!(
+            elapsed >= lower && elapsed <= lower + ms(1),
+            "actual = {:?}, expected = {:?}",
+            elapsed,
+            lower
+        );
+    }};
+}
```

Comment:
this would actually be an awesome macro to expose publicly, I end up adding a utility like this in any test cases I write that track time

Reply:
We already have similar macros in `tokio-test`. I don't see why we couldn't have this one there too